GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeNagTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using GVFS.FunctionalTests.FileSystemRunners;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
+{
+    [TestFixture]
+    [Category(Categories.FullSuiteOnly)]
+    [Category(Categories.Windows)]
+    public class UpgradeNagTests : TestsWithEnlistmentPerFixture
+    {
+        private const string GVFSInstallerName = "SetupGVFS.1.0.18234.1.exe";
+        private const string GitInstallerName = "Git-2.17.1.gvfs.2.5.g2962052-64-bit.exe";
+
+        private string upgradeDirectory;
+        private FileSystemRunner fileSystem;
+
+        public UpgradeNagTests()
+        {
+            this.fileSystem = new SystemIORunner();
+            this.upgradeDirectory = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData, Environment.SpecialFolderOption.Create),
+                "GVFS",
+                "GVFS.Upgrade",
+                "Downloads");
+        }
+
+        [TestCase]
+        [Category(Categories.Windows)]
+        public void NoNagWhenUpgradeNotAvailable()
+        {
+            this.EmptyDownloadDirectory();
+
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "status");
+
+            string.IsNullOrEmpty(result.Errors).ShouldBeTrue();
+        }
+
+        [TestCase]
+        [Category(Categories.Windows)]
+        public void NagWhenUpgradeAvailable()
+        {
+            this.CreateUpgradeInstallers();
+
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "status");
+
+            result.Errors.ShouldContain(new string[] 
+                {
+                    "A newer version of GVFS is available.",

```

Comment:
Forgive me because I can't remember.  Does GVFS do a version comparison when deciding if an upgrade is available?  Or does it simply check if there is anything in the upgrade downloads folder?

Reply:
Yes, GVFS.Upgrader tool does compare actual versions before deciding whether a downloaded upgrade should be installed. GVFS.Hooks does not do detailed version checks only for performance reasons.